coil-base/src/main/java/coil/RealImageLoader.kt

```diff
@@ -268,16 +269,30 @@ internal class RealImageLoader(
     }
 
     /**
-     * Compute the cache key for the [data] + [transformations].
+     * Compute the cache key for the [data] + [parameters] + [transformations].
      */
     private fun <T : Any> computeCacheKey(
         fetcher: Fetcher<T>,
         data: T,
+        parameters: Parameters,
         transformations: List<Transformation>
     ): String? {
         val baseCacheKey = fetcher.key(data) ?: return null
+
         return buildString {
             append(baseCacheKey)
+
+            if (parameters.isNotEmpty()) {
+                for ((key, entry) in parameters) {
+                    val cacheKey = entry.cacheKey ?: continue
+                    append('#')
+                    append(key)
+                    append('=')
+                    append(cacheKey)
+                }
+                append('#')
```

Comment:
You always want the `#` at the end?

Reply:
Just to separate the params and the transformation so we get:

`file:///path/to/file#video_frame=1234#GrayscaleTransformation`

instead of:

`file:///path/to/file#video_frame=1234GrayscaleTransformation`

Probably not required, but it helps visually separate the params.